### PR TITLE
feat: expand calling-rust section

### LINKED ISSUE #2333: [request] Documentation about sending/receiving byte arrays
### Question you want answered

How do you send/receive byte arrays?

### Where did you look for an answer?

All over GitHub primarily, where Lucas explicitly said it's undocumented, and afaik that still holds true.

### Page URL

_No response_

### Additional context

A discord message where I wrote a tiny function that uses it: https://discord.com/channels/616186924390023171/731495028677148753/1254885349725831249
Lucas' comment: https://github.com/tauri-apps/tauri/issues/9322#issuecomment-2029628751

### Are you willing to work on this yourself?

- [ ] I want to work on this myself

### PR DESCRIPTION
**Pull request by lucasfernog**
- expand commands with new functionality (Request, Response, Channel)
- add event system section

closes https://github.com/tauri-apps/tauri-docs/issues/2333

### CODE REVIEW

File: src/content/docs/develop/calling-rust.mdx [hunk 1]
```diff
@@ -323,7 +480,56 @@ pub fn run() {
...
+
+```js
+const data = new Uint8Array([1, 2, 3]);
+await __TAURI__.core.invoke('upload', data, {
+  headers: {
+    Authorization: 'apikey',
+  },
+});
+```
+
```
Comment (vdang-crabnebula) [inline src/content/docs/develop/calling-rust.mdx:531]: We can add a note that  to enable raw request on Linux the feature `linux-ipc-protocol' must be used on the Rust side.

Comment (FabianLars) [reply src/content/docs/develop/calling-rust.mdx:531]: Honestly with how flaky Linux support is i feel we should just raise the minimum webkitgtk version by one and get rid of the feature flag...

Comment (lucasfernog) [reply src/content/docs/develop/calling-rust.mdx:531]: how does that impact the already bad (?) list of supported distros again?

Comment (tweidinger) [reply src/content/docs/develop/calling-rust.mdx:531]: Debain stable seems fine: https://tracker.debian.org/pkg/webkit2gtk
Fedora 40 seems to [deprecate it](https://fedoraproject.org/wiki/Changes/Remove_webkit2gtk-4.0_API_Version ) but still contains it by default in a newer version than required https://packages.fedoraproject.org/pkgs/webkit2gtk4.0/webkit2gtk4.0/ 
Ubuntu 22.04 is also fine (if you update it 😁 ) https://launchpad.net/ubuntu/jammy/+source/webkit2gtk
Other Arch based distros are gonna be fine anyway

Any other major distros I missed?
I think we can bump and remove 👍

**Review by tweidinger — APPROVED** [review]

**Pull request merged**